x/evm/types/genesis.go

```diff
@@ -1,29 +1,217 @@
 package types
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"sort"
+	"strconv"
+	"strings"
 
+	"github.com/axelarnetwork/axelar-core/utils"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+// NewGenesisState returns a new genesis state
+func NewGenesisState(chains []GenesisState_Chain) GenesisState {
+	sort.Slice(chains, less(chains))
+
+	return GenesisState{Chains: chains}
+}
+
+func less(chains []GenesisState_Chain) func(i int, j int) bool {
+	return func(i, j int) bool {
+		return chains[i].Params.Chain < chains[j].Params.Chain
+	}
+}
+
 // DefaultGenesisState returns a default genesis state
 func DefaultGenesisState() *GenesisState {
-	return &GenesisState{DefaultParams()}
+	return &GenesisState{Chains: DefaultChains()}
+}
+
+// DefaultChains returns the default chains for a genesis state
+func DefaultChains() []GenesisState_Chain {
+	p := DefaultParams()
+
+	var chains []GenesisState_Chain
+	for _, params := range p {
+		chain := GenesisState_Chain{
+			Params:            params,
+			BurnerInfos:       nil,
+			CommandQueue:      nil,
+			ConfirmedDeposits: nil,
+			BurnedDeposits:    nil,
+			CommandBatches:    nil,
+			Gateway:           Gateway{},
+			Tokens:            nil,
+		}
+		chains = append(chains, chain)
+	}
+	return chains
```

Comment:
Doesn't this mean we will need to add ethereum on-the-fly?

Reply:
I have the same question here.

Reply:
what do you mean?

Reply:
I mean that we always started the network with Ethereum as part fo the genesis file, with all the bytecodes pre-defined. It looks like we are not doing that anymore, right?

Reply:
it's still there, that has not changed